docs/DesignDocs/InitTimeConfigurationInterface.md

```diff
@@ -0,0 +1,271 @@
+fully reflect the identity of the additional content. For example, `CONFIGID`
+can be set as the hash of the signing key or cert to verify the metadata, and
+`CONFIGSVN` can be set as the version number of the signed metadata.
+
+The scheme of committing the Enclave configuration data into the enclave
+identity produced by the CPU at the enclave initialization time is critical for
+SGX. For the use case of loading additional code/data, `CONFIGID` and
+`CONFIGSVN` reflect the identity of the additional code/data allowed to be
+loaded into the enclave. The identity must be immutable at enclave run time.
+Otherwise, the loaded code/data might be able to spoof its own identity. For
+instance, consider an enclave containing a JavaScript interpreter that executes
+user scripts that originate from the untrusted host in shared enclave memory,
+and which has access to the `oe_get_evidence` API. It is impossible to know
+which script has been executed by such an enclave based on traditional evidence,
+even if the hash of the script is supposed to be included in the
+`REPORT.report_data`, because a malicious script can set `REPORT.report_data`
+and obtain valid evidence via `oe_get_evidence`. Similarly, if an enclave loads
+and executes arbitrary assembly code from the host, this assembly code can use
+the SGX `EREPORT` instruction to create valid evidence reporting a spoofed
+identity of the loaded code. With `CONFIGID` and `CONFIGSVN` set at enclave
+initialization time, and accessible to the enclave code, the static code of the
+enclave makes sure the user script or assembly code loaded into the enclave
+match the identity captured in `CONFIGID` and `CONFIGSVN`, before transferring
+the execution control to the loaded code. Even if the loaded code is malicious,
+the code has no ability to spoof its own identity.  
+
+Enclave Creation Interface
+-------------------------------------
+
+The configurable enclave functionality concept is not limited to SGX TEE. Other
+TEEs can potentially support this concept, including the aspect of immutable
+identity of additional code/code allowed to be loaded into the enclave post
+enclave initialization.
+
+The enclave application passes the Init-time Configuration Data to the enclave
+loader through the `OE_ENCLAVE_INITTIME_CONFIGURATION_DATA` enclave setting
+context:
+
+```C
+/**
+ * Types of settings passed into **oe_create_enclave**
+ */
+typedef enum _oe_enclave_setting_type
+{
+    OE_ENCLAVE_SETTING_CONTEXT_SWITCHLESS = 0xdc73a628,
+#ifdef OE_WITH_EXPERIMENTAL_EEID
+    OE_EXTENDED_ENCLAVE_INITIALIZATION_DATA = 0x976a8f66,
+#endif
+    /** Enclave configuration data committed at Enclave initialization time,
+     *  reflected in TEE-produced enclave identity evidence. Currently only
+     *  supported by SGX Enclaves with KSS feature enabled.
+     */
+    OE_ENCLAVE_INITTIME_CONFIGURATION_DATA = 0x976a8f67,
+} oe_enclave_setting_type_t;
+
+typedef struct _oe_enclave_inittime_configuration_data
```

Comment:
OE coding style is to avoid abbreviations, which means "inittime" should be "initialization_time" or some briefer term but not abbreviation

Reply:
`_oe_enclave_inittime_configuration_data` now removed from the latest version of the PR.